### ISSUE TITLE
arrayItem Property doesn't work for subschemas

### ISSUE THREAD

**Issue by GraemeFulton** [VulcanJS/Vulcan#2626]
In the 1.15 update, the arrayItem property is introduced so we can get rid of the `foo.$` field. I found this doesn't work for subschemas. I get the following error for a subschema defining an urlCollection `arrayItem`:

`Error: "urlCollection" is Array type but the schema does not include a "urlCollection.$" definition for the array items"`

It looks like it doesn't see the arrayItems field. I made the update as outlined in the blog:

<img width="600" alt="Screenshot 2020-08-29 at 12 46 44" src="https://user-images.githubusercontent.com/4354786/91636118-03035c80-e9f6-11ea-8d01-ff04712538f5.png">

It does however work fine for my root schema, but in the nested schemas, I have to continue using the old `foo.$` method.

**Comment by eric-burel** [comment]
Thanks for pointing that out. Will add it whenever we have a chance to rework on schema nesting.

**Comment by GraemeFulton** [comment]
> Thanks for pointing that out. Will add it whenever we have a chance to rework on schema nesting.

Yeah probably not essential at the moment, since the old way still works
